### PR TITLE
ci: add system tests in CI

### PR DESCRIPTION
**Pull request by cbeauchesne**
Add [system tests](https://github.com/DataDog/system-tests) in CI.

System tests is a functional test suite that spawns tracer and agent, run some typical scenario, and then ensure that everything that should happen between tracer/agent/backend really happen. 

As now, the coverage is mainly on appsec features, but almost everything that have a visible effect on tracer/agent surface (*ie* traffic, logs, stdout...) can be tested.

### CODE REVIEW

**Comment by cbeauchesne** [comment]
PR is ready on system tests side. Though, there are 3 failing tests, shall I do something to fix it ?

File: .github/workflows/system-tests.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: System Tests
+
+on:
+  pull_request:
+    branches:
+      - "**"
```
Comment (brettlangdon) [inline .github/workflows/system-tests.yml:6]: Does this have to be defined? I think we can just do `pull_request:` only?

File: .github/workflows/system-tests.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  workflow_dispatch: {}
+  schedule:
+    - cron:  '00 04 * * 2-6'
```
Comment (brettlangdon) [inline .github/workflows/system-tests.yml:9]: Why run on a schedule? Is this required?

File: .github/workflows/system-tests.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+name: System Tests
+
+on:
```
Comment (brettlangdon) [inline .github/workflows/system-tests.yml:3]:
```suggestion
on:
  push:
    branches:
      - master
```

File: .github/workflows/system-tests.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  pull_request:
+    branches:
+      - "**"
+  workflow_dispatch: {}
+  schedule:
+    - cron:  '00 04 * * 2-6'
+
+jobs:
+  system-tests:
+    if: ${{ github.event.pull_request.head.repo.full_name == 'DataDog/dd-trace-py' }}
```
Comment (brettlangdon) [inline .github/workflows/system-tests.yml:13]: This is just to keep it from running (and failing) on forks?

Comment (brettlangdon) [reply .github/workflows/system-tests.yml:13]: Any issues if it did run on a fork and someone had a `DD_API_KEY` set?

File: .github/workflows/system-tests.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - weblog-variant: flask-poc
+          - weblog-variant: uwsgi-poc
```
Comment (brettlangdon) [inline .github/workflows/system-tests.yml:19]: In the future, is this the only thing we'll need to tweak when we add more tests?

File: .github/workflows/system-tests.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Compress artifact
+        if: ${{ always() }}
+        run: tar -czvf artifact.tar.gz $(ls | grep logs)
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: logs_${{ matrix.weblog-variant }}
+          path: artifact.tar.gz
```
Comment (brettlangdon) [inline .github/workflows/system-tests.yml:51]: Can we leave a comment on these two sections that we want to compress and upload the artifacts even on failures.

It is semi-obvious with `always()`, but having a comment just to be sure someone doesn't manually edit and change the expected behavior.

File: .github/workflows/system-tests.yml (hunk 2 above)
Comment (cbeauchesne) [reply .github/workflows/system-tests.yml:9]: Yes it is : the tracer is tested against production version of all other components. If another component is released and introduce an incompatibility, it's the a convenient way to know it.

Further more, all results are collected and displayed on a dashboard, it's helpful to have a global view of all component status. If you do not make any PR, this result can be outdated. (I'll send you the link of the dashboard on slack).

File: .github/workflows/system-tests.yml (hunk 4 above)
Comment (cbeauchesne) [reply .github/workflows/system-tests.yml:13]: > This is just to keep it from running (and failing) on forks?

yes, PR *from* a fork.

That's true, it'll prevent to run it directly on the fork, I'll remove it.

File: .github/workflows/system-tests.yml (hunk 5 above)
Comment (cbeauchesne) [reply .github/workflows/system-tests.yml:19]: More precisely, it's the same test scenario, but on different target app.

Speaking about the devil, @vdeturckheim just added django, I'll add it.

**Review by nizox — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**